src/core_plugins/kbn_vislib_vis_types/public/controls/point_series/series.js

```diff
@@ -57,11 +57,20 @@ module.directive('vislibSeries', function () {
         });
       });
 
+      let horizontal = false;
       $scope.$watch(() => {
         return $scope.vis.params.seriesParams.map(series => series.type).join();
       }, () => {
         const types = _.uniq(_.map($scope.vis.params.seriesParams, 'type'));
         $scope.savedVis.type = types.length === 1 ? types[0] : 'histogram';
+        if (horizontal) $scope.savedVis.type = 'horizontal_bar';
+      });
+
+      $scope.$watch(() => {
+        return $scope.vis.params.categoryAxes[0].position;
+      }, newPosition => {
+        horizontal = (newPosition === 'left' || newPosition === 'right');
+        if (horizontal) $scope.savedVis.type = 'horizontal_bar';
```

Comment:
one issue is that if you do a second switch back from horizontal to vertical, the switch isn't registered. 

we probably want something like:

```
    if (horizontal) {
          $scope.savedVis.type = 'horizontal_bar';
        } else{
          $scope.savedVis.type = 'histogram';
        }
```